main.go

```diff
@@ -13,35 +13,45 @@ import (
 
 	"github.com/go-kit/log"
 	"github.com/go-kit/log/level"
-	"github.com/observatorium/api/client"
-	"github.com/observatorium/api/client/parameters"
 	"github.com/observatorium/obsctl/pkg/config"
+	"github.com/observatorium/obsctl/pkg/fetcher"
 	monitoringv1 "github.com/prometheus-operator/prometheus-operator/pkg/apis/monitoring/v1"
 	monitoringclient "github.com/prometheus-operator/prometheus-operator/pkg/client/versioned"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 	ctrl "sigs.k8s.io/controller-runtime"
 )
 
-const obsctlContext = "api"
+const obsctlContextAPIName = "api"
 const defaultSleepDurationSeconds = 30
 
-var logger log.Logger
+// tenantRulesLoader represents logic for loading and filtering PrometheusRule objects by tenants.
+// Useful for testing without spinning up cluster.
+type tenantRulesLoader interface {
+	GetPrometheusRules() ([]*monitoringv1.PrometheusRule, error)
```

Comment:
do `GetPrometheusRules` and `GetTenantRuleGroups` need to be exported? Looks like they are being used only in `main` pkg?

Reply:
same for the other funcs in this file (e.g. `InitOrReloadObsctlConfig`, `SetCurrentTenant`, etc)

Reply:
Will change!